### PR TITLE
Changed bahavior of getRandom / getRandomKey(s) (Issue #28)

### PR DESCRIPTION
**Pull request by th3l0g4n**
- slightly changed implementation as discussed
- adjusted tests accordingly
- added check in `getRandomValues` as it is affected by the change in `getRandomKeys`

### CODE REVIEW

File: src/AbstractArray.php [hunk 1]
```diff
@@ -491,31 +485,25 @@ public function getRandom()
...
-        $count = $this->count();
-        if ($number < 1 || $number > $count) {
-            throw new \RangeException(sprintf(''
-                . 'The given number "%d" should be greater than or equal to "1" '
-                . 'and less than or equal to "%d" (the number of elements in the array).',
-                $number,
-                $count
-            ));
+        if (!$number || $number === 0) {
+            throw new \LogicException(sprintf('You have to provide a $number greater 0 (%s given)', $number));
```
Comment (bocharsky-bw) [inline src/AbstractArray.php]: I think we need to get rid of `LogicException` and keep only `RangeException` with `if ($number === 0 || $number === $count) {` statement.

**Comment by bocharsky-bw** [comment]
Seems you commited as `Andre` not as `th3l0g4n`. Do you want to change commit author's name? Otherwise, I think this changes not apply to your GitHub account.

**Comment by th3l0g4n** [comment]
I removed the LogicException and adjusted Tests accordingly. Thanks for pointing out the commit info issue, but i`m currently fine if this happens.

File: src/AbstractArray.php [hunk 2]
```diff
@@ -491,31 +485,21 @@ public function getRandom()
...
-                $number,
-                $count
-            ));
-        }
-        if ($number === 1 || $number === $count) {
-            throw new \LogicException(sprintf(
-                'The given number "%d" should not be equal to "1" or "%d" (the number of elements in the array).',
+        if (!$number || $number === 0 ||$number > $count) {
+            throw new \RangeException(sprintf(
+                'Number of requested keys (%s) must be lower or equal than available array elements (%s)',
```
Comment (bocharsky-bw) [inline src/AbstractArray.php]: Could we it change to `... must be equal or lower than number of elements in this array (%s)`?

File: src/AbstractArray.php [hunk 3]
```diff
@@ -491,31 +485,21 @@ public function getRandom()
...
-                . 'The given number "%d" should be greater than or equal to "1" '
-                . 'and less than or equal to "%d" (the number of elements in the array).',
-                $number,
-                $count
-            ));
-        }
-        if ($number === 1 || $number === $count) {
-            throw new \LogicException(sprintf(
-                'The given number "%d" should not be equal to "1" or "%d" (the number of elements in the array).',
+        if (!$number || $number === 0 ||$number > $count) {
```
Comment (bocharsky-bw) [inline src/AbstractArray.php]: `!$number` do not need anymore. `$number` converts to `int` above, so we could simplify it and add empty space after `||` sign:
`if ($number === 0 || $number > $count) {`.

**Comment by bocharsky-bw** [comment]
@th3l0g4n Thanks! My last few comments and I could to merge it.

**Comment by th3l0g4n** [comment]
Ok done. For some strange reason i thought casting strings to int would resolve to false or something :/

**Comment by bocharsky-bw** [comment]
No, in the most cases we get `0` or `1` when casting something to `int`, but we always get integers.

Great! Thank you @th3l0g4n to fixed it!

**Pull request merged**